app/src/main/java/com/example/androidtrainingproject/ui/home/HomeScreen.kt

```diff
@@ -162,6 +229,136 @@ fun Home(navController: NavController) {
+                            modifier = Modifier.padding(top = dimensionResource(id = R.dimen.padding_m)),
+                            style = MaterialTheme.typography.titleSmall
+                        )
+                        RangeSlider(
+                            value = priceRange,
+                            onValueChange = { value ->
+                                priceRange = value.start.roundToInt()
+                                    .toFloat()..value.endInclusive.roundToInt().toFloat()
+                            },
+                            valueRange = 0f..200f,
+                            modifier = Modifier.padding(top = dimensionResource(id = R.dimen.padding_m)),
+                            colors = SliderDefaults.colors(
+                                thumbColor = Purple,
+                                activeTrackColor = Purple,
+                                inactiveTrackColor = Purple,
+                                activeTickColor = Purple,
+                                inactiveTickColor = Purple,
+                                disabledThumbColor = Purple,
+                                disabledActiveTrackColor = Purple,
+                                disabledActiveTickColor = Purple,
+                                disabledInactiveTrackColor = Purple,
+                                disabledInactiveTickColor = Purple
+                            )
+                        )
+                        Row(
+                            modifier = Modifier
+                                .padding(top = dimensionResource(id = R.dimen.padding_sm))
+                                .fillMaxWidth(),
+                            horizontalArrangement = Arrangement.SpaceBetween
+                        ) {
+                            Text(
+                                text = "$${priceRange.start}",
+                                style = MaterialTheme.typography.titleSmall
+                            )
+                            Text(
+                                text = "$${priceRange.endInclusive}",
+                                style = MaterialTheme.typography.titleSmall
+                            )
+                        }
+                        Text(
+                            text = stringResource(id = R.string.product_rating),
+                            modifier = Modifier.padding(top = dimensionResource(id = R.dimen.padding_xl)),
+                            style = MaterialTheme.typography.titleSmall
+                        )
+                        LazyRow(
+                            modifier = Modifier
+                                .padding(top = dimensionResource(id = R.dimen.padding_sm_plus))
+                                .fillMaxWidth(),
+                            horizontalArrangement = Arrangement.SpaceBetween
+                        ) {
+                            items(5) {index ->
+                                if (index < ratingFilter) {
```

Comment:
Always extract list items to a separate composable. This is a rather small view so you can have it in the file that you make for the custom rating bar, right under it.